src/Analyzers/Core/CodeFixes/MakeDeclarationsPartial/AbstractMakeDeclarationsPartialCodeFixProvider.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageService;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.MakeDeclarationsPartial
+{
+    internal abstract class AbstractMakeDeclarationsPartialCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        protected AbstractMakeDeclarationsPartialCodeFixProvider()
+            : base(supportsFixAll: false)
+        {
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var span = context.Span;
+            var cancellationToken = context.CancellationToken;
+
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var syntaxRoot = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var node = syntaxRoot.FindNode(span);
+
+            if (node is null)
+            {
+                Debug.Fail("Cannot find node");
+                return;
+            }
+
+            var declarationName = syntaxFacts.GetIdentifierOfTypeDeclaration(node).ValueText;
```

Comment:
so with compiler diagnostics, you cannot make assumptions about what span/node they may be on.  in other words, FindNode may not actually give youa  TypeDecl back.  Have to check that first before passing into syntaxFacts likek this